jerry-core/parser/js/js-parser-expr.c

```diff
@@ -2653,14 +2693,83 @@ parser_append_binary_single_assignment_token (parser_context_t *context_p, /**<
   return assign_opcode;
 } /* parser_append_binary_single_assignment_token */
 
+/**
+ * Check for invalid Logical operator and Nullish chaining
+ */
+#if ENABLED (JERRY_ESNEXT)
+static void
+parser_check_invalid_logical_op (parser_context_t *context_p, /**< context */
+                                 size_t grouping_level) /**< grouping_level */
+{
+  parser_stack_iterator_t iterator;
+  parser_stack_iterator_init (context_p, &iterator);
+  bool logical_found = (grouping_level & PARSER_GROUPING_LOGICAL_FOUND) != 0;
+  bool nullish_found = false;
+
+  while (true)
+  {
+    uint8_t token = parser_stack_iterator_read_uint8 (&iterator);
+
+    if (token == LEXER_EXPRESSION_START
+        || token == LEXER_LEFT_PAREN
+        || !LEXER_IS_BINARY_OP_TOKEN (token))
+    {
+      return;
+    }
+
+    parser_stack_iterator_skip (&iterator, sizeof (uint8_t));
+
+    if (token == LEXER_ASSIGN)
+    {
+      cbc_opcode_t opcode = (cbc_opcode_t) parser_stack_iterator_read_uint8 (&iterator);
+
+      if (cbc_flags[opcode] & CBC_HAS_LITERAL_ARG)
+      {
+        parser_stack_iterator_skip (&iterator, sizeof (uint16_t));
+      }
+
+      if (JERRY_UNLIKELY (context_p->stack_top_uint8 == LEXER_ASSIGN_GROUP_EXPR
+                          || context_p->stack_top_uint8 == LEXER_ASSIGN_CONST))
```

Comment:
Both can happen at the same time, also accessing the `stack_top_uint8` is incorrect.

```c
token = parser_stack_iterator_read_uint8 (&iterator);

if (token == LEXER_ASSIGN_GROUP_EXPR) { parser_stack_iterator_skip (&iterator, sizeof (uint8_t));}
if (token == LEXER_ASSIGN_CONST) { parser_stack_iterator_skip (&iterator, sizeof (uint8_t));}
```